### ISSUE TITLE
libuv-test-commit-zos is perma-red lately

### ISSUE THREAD

**Issue by cjihrig** [nodejs/build#1949]
Here is an example run: https://ci.nodejs.org/job/libuv-test-commit-zos/1959/nodes=zos-s390x/console.

I *think* this is a build issue, but am happy to be wrong.

cc: @nodejs/libuv

**Comment by sam-github** [comment]
```
% ssh  test-marist-zos13-s390x-1
UNIX1:/u/unix1: > git --version
git version 2.12.0
UNIX1:/u/unix1: >pwd
/u/unix1
UNIX1:/u/unix1: >cd /u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >ls
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >git clone https://github.com/nodejs/node.git
fatal: could not create work tree dir 'node': EDC5111I Permission denied.
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >id
uid=990004(UNIX1) gid=0(IPGROUP)
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >ls -l
total 0
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >ls -l ..
total 32
drwxr-xr-x   3 IOJS     IPGROUP     8192 Oct  9 14:31 zos-s390x
drwxr-xr-x   2 IOJS     IPGROUP     8192 Oct  3 00:05 zos-s390x@tmp
UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >su - iojs
FSUM5019 Enter the password for iojs: 
CEE5206S The signal SIGINT was received.

UNIX1:/u/iojs/build/workspace/libuv-test-commit-zos/nodes/zos-s390x: >sudo su - iojs
sudo: FSUM7351 not found
```

@nodejs/platform-s390  @miladfarca Are you a bit more familiar with Z/os? Any idea what is going on?

**Comment by joransiu** [comment]
Which ID are we suppose to be using `UNIX1` or `iojs`?

**Comment by miladfarca** [comment]
Looking at the console, this seems to be a git-wrapper issue cased by `/u/iojs/git-wrapper`:
```
fatal: bad config file line 1 in /u/iojs/.gitconfig
```

**Comment by mhdawson** [comment]
@sam-github for future reference you need a slightly different command line:

```
su - -s iojs
```

**Comment by sam-github** [comment]
https://ci.nodejs.org/job/libuv-test-commit-zos/nodes=zos-s390x/1962/ passed

**Comment by cjihrig** [comment]
Thanks @sam-github!

**Comment by sam-github** [comment]
Yes, the .gitconfig file is back. I think some of the build shell scripts call git to set the globally configed user name and email... that might be the source of this (I can't think of anything else that would create that file), I'll look into it.

**Comment by sam-github** [comment]
Some

**Comment by sam-github** [comment]
Somewhere, something is calling git directly, not through the wrapper:
```
IOJS:/u/iojs: >/u/iojs/git-wrapper config --global --add core.sparsecheckout yes
IOJS:/u/iojs: >cat .gitconfig
[core]
        sparsecheckout = yes
IOJS:/u/iojs: >git config --global --add core.sparsecheckout yes                
IOJS:/u/iojs: >cat .gitconfig                                   
$?)/,?`ˎ/,?`ˎIOJS:/u/iojs: >git config --global --list                       
fatal: bad config line 1 in file /u/iojs/.gitconfig
```

AFAICT, the only jobs running on that host are libuv-test-commit: https://ci.nodejs.org/computer/test-marist-zos13-s390x-1/builds

So, either the libuv builds are internally, occaisonally calling git directly, or something else is, as the iojs user, or .... black magic?

In the hopes of pinning this down, I've tried to create an empty .gitconfig as unreadable, which will hopefully allow us to find whoever is changing it, becaues the build job has no reason to do so.

But.... no such luck.I guess because git unlinks and replaces the file instead of writing it, making it unwritble doesn't help, and I'm not bold enough to make the iojs home directory unwritable to iojs.

@mhdawson  @miladfarca does anyone or anything else use this machine?

@cjihrig can you think of any reason that the libuv build internally would be calling git?

**Comment by sam-github** [comment]
I tend to think its an external entity, not from inside the job, because it was green for quite a while, then instantly and permanently red.

And for now, I deleted /u/iojs/.gitignore again.

**Comment by cjihrig** [comment]
vcbuild.bat calls git, but that shouldn't impact zOS, right? There is also tools/make_dist_html.py, but that was added back in 2017, so I wouldn't think the problems are related.

**Comment by mhdawson** [comment]
@sam-github I don't know of any other jobs except the libuv ones that use the z/OS machines

**Comment by sam-github** [comment]
I  added `git config --list` as the last command in the build step of https://ci.nodejs.org/job/libuv-test-commit-zos/, it fails if the .gitconfig is corrupt, which should fail the build, and at least give some sense of when the config is damaged.

**Comment by sam-github** [comment]
I can't find where `/u/iojs/git-wrapper` is called from, does anyone know how jenkins knows to call it? Its not in the job config, so some other part of jenkins must have been customized to know to use it. If possible, it would be useful to modify it. The most direct would be to simply `rm /u/iojs/.gitconfig` in its beginning, because the file should never be present. Adding a call to `git config --list` at the end just to see if its the culprit in damaging the file would also be possible.

**Comment by mhdawson** [comment]
@sam-github I thought you were on CC on a different issue/thread where we confirmed that /u/iojs/git-wrapper is configured through the tools section as what is used by default for git/ (but maybe not, not sure which issue that was)

![image](https://user-images.githubusercontent.com/9373002/67320231-ea740d00-f4db-11e9-812e-560a6069e7f2.png)

**Comment by miladfarca** [comment]
Just noticed our zos machine still has the same issue on a libuv PR: 
https://ci.nodejs.org/job/libuv-test-commit-zos/1986/console

**Comment by sam-github** [comment]
I removed the binary .gitconfig files again.

And I made the iojs home directory unwritable for itself.

- before: `drwxrwxr-x  10 IOJS     IPGROUP     8192 Oct 23 15:59 iojs`
- `chmod u-w /u/iojs` (as iojs user)
- after: `dr-xrwxr-x  10 IOJS     IPGROUP     8192 Oct 23 16:01 /u/iojs`

Now attempts to damage the .gitconfig fail:
```
IOJS:/u/iojs: >git config --global --add core.sparsecheckout yes
error: could not lock config file /u/iojs/.gitconfig: EDC5111I Permission denied.
```

I'm hoping that our build jobs don't have a need to write anything into the home directory, I'll see if this passes:
- https://ci.nodejs.org/job/libuv-test-commit-zos/1988/nodes=zos-s390x/console

If it does pass, I'll leave the home directory unwritable, in the hopes of catching whatever is running raw `git config` commands.
